modules/mi_datagram/mi_datagram_writer.c

```diff
@@ -65,7 +65,7 @@ static inline int mi_datagram_write_node(datagram_stream * dtgram,
 
 
 	start = p = dtgram->current;
-	end = dtgram->start + dtgram->len;
+	end = dtgram->current + dtgram->len;
```

Comment:
I'm not sure this is correct - the `dtgram` has a buffer of a fixed `len` (starting with `start`). The `current` is the offset (inside the buffer) with the writing. And we need the `end` as end of the buffer to prevent the overflow.  The end of the buffer is the same all the time (start + end), disregarding where the writing offset is.

Reply:
ok, sorry, I see, the `len` is the remaining len, not the total len...you are right here :+1: